### PR TITLE
Expose `containerPort` as `$PORT_NAME` env vars

### PR DESCRIPTION
**Pull request by zen-dog**
when using docker user networking (and not `hostPorts` exist). This part of code went missing from 1.3 to 1.4.

JIRA: COPS-2072

### CODE REVIEW

File: src/main/scala/mesosphere/mesos/EnvironmentHelper.scala [hunk 1]
```diff
@@ -87,10 +87,11 @@ object EnvironmentHelper {
           env += (s"PORT_$generatedPort" -> generatedPort.toString)
       }
 
-      requestedPorts.map(_.name).zip(effectivePorts).foreach {
-        case (Some(portName), Some(effectivePort)) =>
+      requestedPorts.zip(effectivePorts).foreach {
+        case (PortRequest(Some(portName), _), Some(effectivePort)) =>
           env += (s"PORT_${portName.toUpperCase}" -> effectivePort.toString)
-        // TODO(jdef) port name envvars for generated container ports
+        case (PortRequest(Some(portName), port), None) =>
```
Comment (jdef) [inline src/main/scala/mesosphere/mesos/EnvironmentHelper.scala]: same comments as for #5887

File: src/test/scala/mesosphere/mesos/TaskBuilderTest.scala [hunk 2]
```diff
@@ -1630,6 +1630,36 @@ class TaskBuilderTest extends UnitTest {
...
+          envPrefix = None
+        )
+
+      val env: Map[String, String] =
+        command.getEnvironment.getVariablesList.toList.map(v => v.getName -> v.getValue).toMap
+
+      assert("8080" == env("PORT_8080"))
+      assert("8080" == env("PORT_HTTP"))
+
+      val port1 = env("PORT1")
```
Comment (jdef) [inline src/test/scala/mesosphere/mesos/TaskBuilderTest.scala]: ditto re: sanity check that `port1 ! = 0`

File: src/test/scala/mesosphere/mesos/TaskBuilderTest.scala [hunk 3]
```diff
@@ -1,25 +1,25 @@
...
+import mesosphere.marathon.state.{AppDefinition, Container, PathId, Timestamp, _}
 import mesosphere.marathon.stream.Implicits._
-import mesosphere.marathon.test.{ MarathonTestHelper, SettableClock }
-import mesosphere.marathon.{ Protos, _ }
-import mesosphere.mesos.protos.{ Resource, _ }
+import mesosphere.marathon.test.{MarathonTestHelper, SettableClock}
+import mesosphere.marathon.{Protos, _}
+import mesosphere.mesos.protos.{Resource, _}
 import org.apache.mesos.Protos.TaskInfo
-import org.apache.mesos.{ Protos => MesosProtos }
```
Comment (meln1k) [inline src/test/scala/mesosphere/mesos/TaskBuilderTest.scala]: Looks like intelij has different opinion on how to format imports compared to SBT :)

**Comment by jdef** [comment]
lgtm

the port-by-name docs could use some clarification/editing as well: https://github.com/mesosphere/marathon/blob/master/docs/docs/networking.md#per-task-environment-variables

for example, PORT_name doesn't always refer to a host port. these variables are actually pretty confusing because their meaning changes depending upon how ports are allocated by an app. It might be worth a table that clearly shows how the value of the PORTxxx variables changes depending on the app configuration.

**Pull request merged**